flutter/shell/platform/tizen/tizen_renderer_vulkan.cc

```diff
@@ -0,0 +1,972 @@
+}
+
+FlutterVulkanPhysicalDeviceHandle
+TizenRendererVulkan::GetPhysicalDeviceHandle() {
+  return physical_device_;
+}
+
+FlutterVulkanDeviceHandle TizenRendererVulkan::GetDeviceHandle() {
+  return logical_device_;
+}
+
+uint32_t TizenRendererVulkan::GetQueueIndex() {
+  return graphics_queue_family_index_;
+}
+
+const char** TizenRendererVulkan::GetEnabledInstanceExtensions() {
+  return enabled_instance_extensions_.data();
+}
+
+const char** TizenRendererVulkan::GetEnabledDeviceExtensions() {
+  return enabled_device_extensions_.data();
+}
+
+void* TizenRendererVulkan::GetInstanceProcAddress(
+    FlutterVulkanInstanceHandle instance,
+    const char* name) {
+  return reinterpret_cast<void*>(
+      vkGetInstanceProcAddr(reinterpret_cast<VkInstance>(instance), name));
+}
+
+FlutterVulkanImage TizenRendererVulkan::GetNextImage(
+    const FlutterFrameInfo* frameInfo) {
+  if (resize_pending_) {
+    RecreateSwapChain();
+  }
+  VkResult result;
+  do {
+    result = vkAcquireNextImageKHR(logical_device_, swapchain_, UINT64_MAX,
+                                   VK_NULL_HANDLE, image_ready_fence_,
+                                   &last_image_index_);
+    if (result == VK_ERROR_OUT_OF_DATE_KHR) {
+      RecreateSwapChain();
+    } else if (result == VK_SUBOPTIMAL_KHR) {
+      break;
+    }
+  } while (result != VK_SUCCESS);
+
+  vkWaitForFences(logical_device_, 1, &image_ready_fence_, VK_TRUE, UINT64_MAX);
+  vkResetFences(logical_device_, 1, &image_ready_fence_);
+  return {
+      .struct_size = sizeof(FlutterVulkanImage),
+      .image = reinterpret_cast<uint64_t>(swapchain_images_[last_image_index_]),
+      .format = static_cast<uint32_t>(surface_format_.format),
+  };
+}
+
+bool TizenRendererVulkan::Present(const FlutterVulkanImage* image) {
+  VkPipelineStageFlags stage_flags =
+      VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+  VkSubmitInfo submit_info{};
+  submit_info.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO;
+  submit_info.waitSemaphoreCount = 0;
+  submit_info.pWaitSemaphores = VK_NULL_HANDLE;
+  submit_info.pWaitDstStageMask = &stage_flags;
+  submit_info.commandBufferCount = 1;
+  submit_info.pCommandBuffers = &present_transition_buffers_[last_image_index_];
+  submit_info.signalSemaphoreCount = 1;
+  submit_info.pSignalSemaphores = &present_transition_semaphore_;
+  vkQueueSubmit(graphics_queue_, 1, &submit_info, VK_NULL_HANDLE);
+
+  VkPresentInfoKHR present_info{};
+  present_info.sType = VK_STRUCTURE_TYPE_PRESENT_INFO_KHR;
+  present_info.waitSemaphoreCount = 1;
+  present_info.pWaitSemaphores = &present_transition_semaphore_;
+  present_info.swapchainCount = 1;
+  present_info.pSwapchains = &swapchain_;
+  present_info.pImageIndices = &last_image_index_;
+  VkResult result = vkQueuePresentKHR(graphics_queue_, &present_info);
+
+  if (result == VK_SUBOPTIMAL_KHR || result == VK_ERROR_OUT_OF_DATE_KHR) {
+    RecreateSwapChain();
+  }
+  vkDeviceWaitIdle(logical_device_);
+  return result == VK_SUCCESS;
```

Comment:
This function returns false after RecreateSwapChain() when result is VK_SUBOPTIMAL_KHR or VK_ERROR_OUT_OF_DATE_KHR. Is this intended?